pkg/engine/params_k8s.go

```diff
@@ -28,10 +28,12 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 		k8sComponents := api.K8sComponentsByVersionMap[k8sVersion]
 		kubernetesConfig := orchestratorProfile.KubernetesConfig
 		kubernetesImageBase := kubernetesConfig.KubernetesImageBase
-		hyperKubeImageBase := kubernetesConfig.KubernetesImageBase
+		hyperkubeImageBase := kubernetesConfig.KubernetesImageBase
+		hyperkubeComponent := k8sComponents["hyperkube"]
 
 		if properties.IsAzureStackCloud() {
 			kubernetesImageBase = cloudSpecConfig.KubernetesSpecConfig.KubernetesImageBase
+			hyperkubeComponent = k8sComponents["hyperkube-azs"]
```

Comment:
this only exists for 1.11 right now so we should only override if the k8s version is 1.11.x

Reply:
I'll wait for you to confirm that a new map entry for `hyperkube-azs` is the preferred approach before I make changes here.

If that's the case, I am thinking on adding the entry to all `K8sVersionComponents` maps.
However, that implies that only versions supported by AZS should be accepted.
This validation may tilt the scale for an approach that requires new entries for map `GetAllSupportedKubernetesVersions`.